src/modals/NewAddressModal.tsx

```diff
@@ -59,7 +59,12 @@ const NewAddressModal = ({ title, onClose, singleAddress }: NewAddressModalProps
       setNewAddressData(address)
       setNewAddressGroup(addressToGroup(address.hash, TOTAL_NUMBER_OF_GROUPS))
     }
-  }, [generateAddress, singleAddress])
+    // Without disabling eslint, we need to add `generateAddress` in the deps. Doing so results in infinite renders,
+    // even after wrapping it in a useCallback. The only solution would be to implement generateAddress in this
+    // component and wrap it in useCallback. Which might not be a bad idea since it's not used anywhere else. But then
+    // we don't have a unique place for all address generation function. Which is also fine.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [singleAddress])
```

Comment:
Bringing your attention here. I'd rather disable the linter in this case.

Reply:
The reason why we generate the address when the user opens the modal is so that we know with what value to populate the "group" field. Also, we are not using web workers for generating a single address (yet). Potential improvements:
- Use web workers for all address generation functions
- Generate address only on user "Generate" click
- Initialize the "group" field with `undefined` instead of the actual group of the automatically-generated address

Reply:
Yep! FYI I've tried to come up with a "hack" to prevent the infinite loop (ie. storing the already generated address in a ref, to avoid 2 successive generations) but disabling eslint is def. easier for now :)